ui-v2/tests/unit/helpers/token/is-legacy-test.js

```diff
@@ -44,3 +44,10 @@ test('it returns false if tokens have no Rules, or has Rules but those rules are
   const actual = isLegacy([[{}, { Rules: '' }]]);
   assert.notOk(actual);
 });
+test('it returns false if a token is marked as legacy, has Rules but those rules are empty', function(assert) {
+  // this may seem strange, but empty Rules should override Legacy, this only happens
+  // when a legacy token that has already been loaded has its rules wiped out
+  // WITHOUT then the ui refreshing
```

Comment:
What happens when the UI refreshes?

Reply:
When the UI refreshes (so a full page/browser refresh), it asks the API again. The token that you saved with with empty rules (so `Rules: '', Legacy: true`) and therefore in 'non-legacy' limbo, is pulled back in via the API. As that token now has no Rules the backend then gives it to me as `Legacy: false` and all is good, well it doesn't even give me that property anymore.

I could have set `Legacy` to equal false myself, but both the frontend and backend use `Legacy` as little as possible, and it will eventually go away. It's only on the listing that its used as we decided not to send `Rules` in the listing API response.